File: java/src/main/java/io/fivexer/sdk/model/PushSubscription.java
package io.fivexer.sdk.model;

/**
 * A registered browser Web Push subscription.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class PushSubscription {
    private String endpoint;
    private String createdAt;

    public String getEndpoint() { return endpoint; }
    public String getCreatedAt() { return createdAt; }
}

File: php/src/Model/PushSubscription.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A registered browser Web Push subscription.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class PushSubscription
{
    public function __construct(
        public readonly string $endpoint,
        public readonly string $createdAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            endpoint: (string) ($data['endpoint'] ?? ''),
            createdAt: (string) ($data['createdAt'] ?? ''),
        );
    }
}
